### ISSUE TITLE
Question on usage for specific multiclass classification use case

### ISSUE THREAD

**Issue by alexHeu** [flairNLP/flair#827]
Hi,

first of all, thanks for providing this great library to the community. My question is of conceptual nature and I was wondering if you could give me some tips or best practices on how to tackle the problem that I am working on right now:
The aim of my project is to identify potential symptoms of a home appliance, e.g. a dishwasher, in texts that is generated by call center agents that take calls from customers that have a problem.
There is a list of roughly 50 predefined symptoms that may appear in a device and the task is to map the text to a symptom. It can be considered as multiclass problem since most of the time only one symptom is available.
The texts are in german, short (usually 1-3 sentences), contain a large number of spelling mistakes and some very domain specific language and abbreviations.
There are 150k texts in total and I have labels for about 10k of them.

My first approach was to train FastText embeddings (since they are quite robust to spelling mistakes), aggregating them to a document vector via taking the mean and building a multiclass classifier on top of the document embeddings to predict the label. For some classes this approach workes quite well but for others unfortunately it does not (e.g. longer texts that contain unimportant information). It seems that by taking the mean to generate the document embedding, the important parts of the texts get overshadowed by the unimportant parts. 
Therefore, I was thinking of a different approach utilizing your library:
1.	Training custom flair embeddings on my corpus
2.	Use DocumentRNNEmbeddings	 consisting of the pretrained ‘de-X’ Flair embeddings, 'bert-base-multilingual-cased' Bert embeddings and the custom trained Flair embeddings
3.	 Train a text classifier

Do you think that this approach is reasonable? More specifically:
-	Can 150k texts consisting of 1-3 sentences be enough to train reasonable custom embeddings? 
-	Are the pre-trained embeddings capable of handling spelling mistakes similar to FastText?
-	Would it make sense to also include other types of embeddings, e.g. CharacterEmbeddings?
-	As I understood it, the downstream training tasks only adapts the DocumentRNNEmbeddings but not the initial word embeddings themselves?
-	Are there any hyper-parameters that are especially important?

I know that some of these things might be answered by just trying it out and experimenting a bit. Nevertheless I would be very grateful if you could share some insights about your experience with these kinds of tasks.

Thanks a lot
Alex

**Comment by stefan-it** [comment]
Hi @alexHeu,

to your questions:

* Could you also specify the number of tokens? I did experiments with low resource languages and even with only 36M tokens, Flair embeddings are really improving results on downstream tasks like PoS tagging
* I would say yes, because the underlying language model is character based, so it should handle spelling mistakes. Btw. here's a paper from my research group regarding to improve NER on Historic German and we also face problems with "noisy" OCR (spelling mistakes are also a problem...). You can find the paper [here](https://arxiv.org/abs/1906.07592)
* You could also use Word Embeddings in combination with the `DocumentRNNEmbeddings` class. You can also think of training your own word embeddings from scratch on your corpus to see, if that improves results
* I would say you should play around with `batch_size` and different word embeddings types. Additionally, if you choose to use BERT just run some experiments with the cased and uncased model (incl. some per-layer analysis).
